functions/src/resolve-market.ts

```diff
@@ -221,4 +201,34 @@ const sendResolutionEmails = async (
   )
 }
 
+function getResolutionParams(outcomeType: string, body: string) {
+  if (outcomeType === 'NUMERIC') {
+    return {
+      ...validate(numericSchema, body),
+      resolutions: undefined,
+      probabilityInt: undefined,
+    }
+  } else if (outcomeType === 'FREE_RESPONSE') {
+    const { outcome, resolutions } = validate(freeResponseSchema, body)
+    if (!(outcome === 'MKT' && resolutions))
+      throw new APIError(
+        400,
+        'Resolutions cannot be specified with MKT outcome'
+      )
```

Comment:
This isn't right. The corresponding old code was this:
```typescript
 if (isNaN(+outcome) && !(outcome === 'MKT' && resolutions) && outcome !== 'CANCEL')
  // .. error case
```
That is, it's trying to express that either the outcome must be a numeric response index, 'MKT' with multiple resolved numeric indices, or 'CANCEL'. It expressed that extremely badly, but your version expresses it even worse :-)

It would be nice to use some combination of `zod` and manual checks here to actually do better than the old code, e.g. check that the indices correspond to valid answers. But at least it has to accept all legal resolutions.